zebra-network/src/meta_addr.rs

```diff
@@ -160,135 +823,244 @@ impl MetaAddr {
+    pub fn has_been_attempted(&self) -> bool {
+        self.get_last_attempt().is_some()
     }
 
     /// Is this address a directly connected client?
+    ///
+    /// If we don't have enough information, assume that it is not a client.
     pub fn is_direct_client(&self) -> bool {
         match self.last_connection_state {
-            Responded => !self.services.contains(PeerServices::NODE_NETWORK),
-            NeverAttemptedGossiped | NeverAttemptedAlternate | Failed | AttemptPending => false,
+            Responded { services, .. } => !services.contains(PeerServices::NODE_NETWORK),
+            NeverAttemptedSeed
+            | NeverAttemptedGossiped { .. }
+            | NeverAttemptedAlternate { .. }
+            | Failed { .. }
+            | AttemptPending { .. } => false,
         }
     }
 
     /// Is this address valid for outbound connections?
+    ///
+    /// If we don't have enough information, assume that it is valid.
     pub fn is_valid_for_outbound(&self) -> bool {
-        self.services.contains(PeerServices::NODE_NETWORK)
+        self.get_untrusted_services()
+            .map(|services| services.contains(PeerServices::NODE_NETWORK))
+            .unwrap_or(true)
             && !self.addr.ip().is_unspecified()
             && self.addr.port() != 0
     }
 
-    /// Return a sanitized version of this `MetaAddr`, for sending to a remote peer.
-    pub fn sanitize(&self) -> MetaAddr {
+    /// Return a sanitized version of this [`MetaAddr`], for sending to a remote peer.
+    ///
+    /// If [`None`], this address should not be sent to remote peers.
+    pub fn sanitize(&self) -> Option<MetaAddr> {
+        // Sanitize the time
         let interval = crate::constants::TIMESTAMP_TRUNCATION_SECONDS;
-        let ts = self.get_last_seen().timestamp();
-        let last_seen = Utc.timestamp(ts - ts.rem_euclid(interval), 0);
-        MetaAddr {
+        let ts = self.get_last_success_or_untrusted()?.timestamp();
+        let last_seen_maybe_untrusted = Utc.timestamp(ts - ts.rem_euclid(interval), 0);
+
+        Some(MetaAddr {
             addr: self.addr,
-            // services are sanitized during parsing, or set to a fixed valued by
-            // new_local_listener, so we don't need to sanitize here
-            services: self.services,
-            last_seen,
-            // the state isn't sent to the remote peer, but sanitize it anyway
-            last_connection_state: NeverAttemptedGossiped,
-        }
+            // the exact state variant isn't sent to the remote peer, but sanitize it anyway
+            last_connection_state: NeverAttemptedGossiped {
+                untrusted_last_seen: last_seen_maybe_untrusted,
+                // services are sanitized during parsing, or set to a fixed value by
+                // new_local_listener, so we don't need to sanitize here
+                untrusted_services: self.get_untrusted_services()?,
+            },
+        })
     }
```

Comment:
Security: Zebra should stop gossiping failure times as last_seen times #1868

Part 3: Sanitize `PeerAddrState` before sending it to other peers, skipping peers with missing fields
* if the peer doesn't have a last seen time, skip it
  * this only happens for seed peers without any success, gossip, or alternate changes
* truncate the last seen time